test/suite/test_cursor_bound19.py

```diff
@@ -0,0 +1,270 @@
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest, ctypes
+from wtscenario import make_scenarios
+from wtbound import bound_base
+
+# test_cursor_bound18.py
+#    Basic cursor bound API\ validation.
+class test_cursor_bound18(bound_base):
+    file_name = 'test_cursor_bound18'
+    use_index = True
+
+    
+    types = [
+        ('table', dict(uri='table:', use_colgroup=False)),
+        ('colgroup', dict(uri='table:', use_colgroup=True))
+    ]
+
+    key_formats = [
+        ('string', dict(key_format='S')),
+        ('var', dict(key_format='r')),
+        ('int', dict(key_format='i')),
+        ('bytes', dict(key_format='u')),
+        ('composite_string', dict(key_format='SSS')),
+        ('composite_int_string', dict(key_format='iS')),
+        ('composite_complex', dict(key_format='iSru')),
+    ]
+
+    value_formats = [
+        ('string', dict(value_format='S')),
+        ('int', dict(value_format='i')),
+        ('bytes', dict(value_format='u')),
+        ('composite_string', dict(value_format='SSS')),
+        ('composite_int_string', dict(value_format='iS')),
+        ('composite_complex', dict(value_format='iSru')),
+    ]
+
+    config = [
+        ('no-evict', dict(evict=False)),
+        ('evict', dict(evict=True))
+    ]
+
+    def set_bounds(self, cursor, key, bound_config, inclusive=True):
+        inclusive_config = ""
+        if (not inclusive):
+            inclusive_config = ",inclusive=false"
+        # Set key and bounds.    
+        cursor.set_key(key)
+        return cursor.bound("bound={0}{1}".format(bound_config, inclusive_config))
+
+    def gen_index_create_param(self):
+        create_params = ",columns=("
+        start = 0
+        for _ in self.key_format:
+            create_params += "k{0},".format(str(start)) 
+            start += 1
+
+        start = 0
+        for _ in self.value_format:
+            create_params += "v{0},".format(str(start)) 
+            start += 1
+        create_params += ")"
+
+        if (self.use_colgroup):
+            create_params += ",colgroups=("
+            start = 0
+            for _ in self.value_format:
+                create_params += "g{0},".format(str(start)) 
+                start += 1
+            create_params += ")"
+        return create_params
+
+    def check_val(self, i):
```

Comment:
Because index tables play on the generated values I wanted to check against the value format not the key format. I'll try can add a new argument to add in a custom format.